### PR TITLE
Add support for gateway as oauth-resourceserver

### PR DESCRIPTION
**Pull request by tgeens**

### CODE REVIEW

File: src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java [hunk 1]
```diff
@@ -179,6 +186,8 @@ public SecurityWebFilterChain springWebFilterChain(
             http.formLogin();
         }
 
+        http.cors(cors -> cors.configurationSource(exchange -> new CorsConfiguration().applyPermitDefaultValues()));
```
Comment (tgeens) [inline src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java:189]: This was for testing, should be removed.
(Integration tests DO succeed without CORS config)

Comment (tgeens) [reply src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java]: Removed

**Comment by tgeens** [comment]
@dependabot rebase.

File: src/main/resources/application-console.yml [hunk 2]
```diff
@@ -14,10 +14,9 @@ spring:
     gateway:
       routes:
         - id: architect
-          uri: http://${DOCKER_HOST_IP:172.17.0.1}:9083/
+          uri: http://${DOCKER_HOST_IP:172.17.0.1}:8083/
```
Comment (vierbergenlars) [inline src/main/resources/application-console.yml:17]: Hmm, why was this changed?

File: src/test/java/eu/xenit/alfred/content/gateway/security/OidcAuthenticationIntegrationTest.java [hunk 3]
```diff
@@ -87,14 +101,14 @@ public void keycloakOIDC_redirectFlow() {
         var initialRedirect = initialResponse.getResponseHeaders().getLocation();
 
         var redirectToKeycloakResponse = rest.get()
-                .uri(Objects.requireNonNull(baseUri.resolve(initialRedirect)))
+                .uri(Objects.requireNonNull(URI.create("http://localhost:" + port).resolve(initialRedirect)))
```
Comment (vierbergenlars) [inline src/test/java/eu/xenit/alfred/content/gateway/security/OidcAuthenticationIntegrationTest.java:104]: Wasn't it better to use a variable here instead of inlining the value everywhere?
Instead of `baseUri`, you can call it `gatewayBaseUri`

**Review by rschev — APPROVED** [review]

File: src/main/resources/application-console.yml (hunk 2 above)
Comment (rschev) [reply src/main/resources/application-console.yml:17]: It's not application-bootRun.yml

Comment (tgeens) [reply src/main/resources/application-console.yml:17]: It's a custom profile, for when you would work on contentcloud-gateway in the context of the contentcloud-api. I would expect that you usually would run the rest of the stack with compose.

File: src/test/java/eu/xenit/alfred/content/gateway/security/OidcAuthenticationIntegrationTest.java (hunk 3 above)
Comment (tgeens) [reply src/test/java/eu/xenit/alfred/content/gateway/security/OidcAuthenticationIntegrationTest.java]: Ack, resolved
